src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -420,10 +420,10 @@ class RideConstructionWindow final : public Window
         }
         switch (_currentTrackCurve)
         {
-            case TRACK_CURVE_LEFT_VERY_SMALL:
-            case TRACK_CURVE_LEFT_SMALL:
-            case TRACK_CURVE_LEFT:
-            case TRACK_CURVE_LEFT_LARGE:
+            case EnumValue(TrackCurve::LeftVerySmall):
+            case EnumValue(TrackCurve::LeftSmall):
+            case EnumValue(TrackCurve::Left):
+            case EnumValue(TrackCurve::LeftLarge):
```

Comment:
It would be a lot easier to cast the `_currentTrackCurve` variable to the enum on the `switch`, then the cases would become clearer and more direct.

On another note, is there any reason for not changing `currentTrackCurve` to be of the enum type?

Reply:
The problem with that is that `_currentTrackCurve` is often _not_ one of the enum types. It is either a curve type OR a specific track type ID. A Google search suggest that casting an invalid value to the enum class results in undefined behaviour, so it kinda has to be done this way, until `_currentTrackCurve` is completely refactored.

Reply:
Oh, that's unfortunate.